### PR TITLE
Refactor Document JSON put format doc

### PR DESCRIPTION
**Pull request by vekterli**
@geirst @kkraune please review

Add SD example for each documented type and disambiguate namespace vs
doctype in document IDs.

### CODE REVIEW

File: documentation/reference/document-json-put-format.html [hunk 1]
```diff
@@ -49,19 +77,27 @@ <h3 id="array-values">Array values</h3>
 <h3 id="map-values">Map values</h3>
 <p>
 Maps are represented as JSON dictionaries. The key has to be a string.
```
Comment (jobergum) [inline documentation/reference/document-json-put-format.html]: The key in the json representation or in the Vespa document definition? The example points in the direction of the former.

Comment (vekterli) [reply documentation/reference/document-json-put-format.html]: It's the former, as that's the only thing supported by JSON. Will rewrite to explicitly mention this.

File: documentation/reference/document-json-put-format.html [hunk 2]
```diff
@@ -127,9 +195,17 @@ <h3 id="tensor">Tensor values</h3>
 <p>
 Feeding a <a href="search-definitions-reference.html#type:tensor">tensor</a> field
 is done by specifying the cells of the tensor as follows:
+</p>
+<pre>
+document tensordoctype1 {
+  field tensorfield type tensor(x{},y{}) {
+    indexing: attribute | summary
+  }
```
Comment (geirst) [inline documentation/reference/document-json-put-format.html]: We also need: "attribute: tensor(x{},y{})"

File: documentation/reference/document-json-put-format.html [hunk 3]
```diff
@@ -140,14 +216,22 @@ <h3 id="tensor">Tensor values</h3>
...
 </p><p>
 In the next example we have a tensor with two indexed dimensions <em>x</em> and <em>y</em>
-representing a 2x2 matrix. The tensor type in this case is <em>tensor(x[2],y[2])</em>.</p>
+representing a 2x2 matrix.
+</p>
+<pre>
+document tensordoctype2 {
+  field tensorfield type tensor(x[2],y[2]) {
+    indexing: attribute | summary
+  }
```
Comment (geirst) [inline documentation/reference/document-json-put-format.html]: We also need: "attribute: tensor(x[2],y[2])"

**Comment by vekterli** [comment]
@geirst PTAL

**Pull request merged**